assets/src/hooks/useDetour.ts

```diff
@@ -54,6 +61,14 @@ export const useDetour = ({ routePatternId, shape }: OriginalRoute) => {
     }
   }, [routePatternId, startPoint, endPoint])
 
+  useEffect(() => {
+    if (startPoint) {
+      fetchNearestIntersection(startPoint.lat, startPoint.lon).then((result) =>
+        setNearestIntersection(result)
+      )
+    }
+  }, [startPoint])
```

Comment:
suggestion: could you use `useNearestIntersection` instead of `fetchNearestIntersection`?

Reply:
I considered it, but was first running into issues, and subsequently figured the hook wasn't what I needed here anyway.

First the issues: fetching the nearest intersection isn't valid until the first startPoint is placed. But conditionally running the hook is not permissible. So I could change the innards of the custom hook to skip running if the lat/lon are null.

Then I figured that the benefit of this hook is that it forms a sort of continual listener, given a dynamic vehicle position. Since placed waypoint is not dynamic, I decided that just doing a standalone fetch was preferred. What do you think?

Reply:
This might be a super basic question, but I want to learn! I'm still a bit stuck on the second part of my comment, on why a hook is better here than the singular fetch? In this case, `nearestIntersection` is only dependent on startPoint and won't be dynamic, similar to how `finishedDetour` works. And in that `useEffect`, we directly fetch the finished detour status without a hook

Reply:
So we don't _have_ to use `useNearestIntersection` but you've essentially just implemented it[`useNearestIntersection`] here, but with less guard rails around `useEffect` and having the state now be managed by `useDetour` instead of `useNearestIntersection`.

Reply:
I thought this would be incredibly straightforward, because I actually had a branch with much of this work done (hence I planned to push the changes first thing yesterday.) Until I realized I needed a lot of test fixes, and sadly thats what I've been pulling my hair out over, yet again. 😓 Pushing up my progress thus far to see if at least the hook restructuring is on the right track. And putting it in a separate PR: https://github.com/mbta/skate/pull/2556